src/app/child-dev-project/children/child-details/form/form.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, Input, OnChanges, SimpleChanges } from "@angular/core";
+import { Child } from "../../model/child";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { EntityMapperService } from "../../../../core/entity/entity-mapper.service";
+import { AlertService } from "../../../../core/alerts/alert.service";
+import { ChildPhotoService } from "../../child-photo-service/child-photo.service";
+import { Router } from "@angular/router";
+import { SessionService } from "../../../../core/session/session-service/session.service";
+
+@Component({
+  selector: "app-form",
+  templateUrl: "./form.component.html",
+  styleUrls: ["./form.component.scss"],
+})
+export class FormComponent implements OnChanges {
+  @Input() child: Child;
+
+  creatingNew = false;
+  isAdminUser: boolean;
+  enablePhotoUpload = false;
+
+  editing: boolean = false;
+  form: FormGroup;
+  validateForm: boolean = false;
+  config;
+
+  constructor(
+    private fb: FormBuilder,
+    private entityMapperService: EntityMapperService,
+    private alertService: AlertService,
+    private childPhotoService: ChildPhotoService,
+    private router: Router,
+    private sessionService: SessionService
+  ) {
+    this.isAdminUser = this.sessionService.getCurrentUser().admin;
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.hasOwnProperty("child")) {
+      this.initForm();
+    }
+  }
+
+  onInitFromDynamicConfig(config: any) {
+    this.child = config.child;
+    this.config = config.config;
+    this.initForm();
+    if (!config.child.name) {
+      this.creatingNew = true;
+      this.switchEdit();
+    }
+  }
+
+  switchEdit() {
+    this.editing = !this.editing;
+    this.initForm();
+    this.enablePhotoUpload = this.childPhotoService.canSetImage();
+  }
+
+  async save(): Promise<Child> {
+    this.checkFormValidity();
+    this.assignFormValuesToChild(this.child, this.form);
+    try {
+      await this.entityMapperService.save<Child>(this.child);
+      if (this.creatingNew) {
+        this.router.navigate(["/child", this.child.getId()]);
+      }
+      this.alertService.addInfo("Saving Successful");
+      this.switchEdit();
+      return this.child;
+    } catch (err) {
+      this.alertService.addDanger(
+        'Could not save Child "' + this.child.name + '": ' + err
```

Comment:
can we get the form panel title as context for the log message?

Reply:
Currently it is not passed, because the title is set by the surrounding component. This way we can ensure a general styling for the titles. 
The alternative approach would be to let each component take care of displaying an title itself.

I understand that you want to show the user where this alert is coming from, but because it is shown after the user clicks the save button of a specific form-component, I think the user understands that the error is connected with this form.